src/widget/wsingletoncontainer.cpp

```diff
@@ -0,0 +1,61 @@
+#include "widget/wsingletoncontainer.h"
+
+#include <QLayout>
+
+#include "util/assert.h"
+#include "skin/skincontext.h"
+#include "widget/wlibrary.h"
+
+
+WSingletonContainer::WSingletonContainer(QWidget* pParent)
+        : WWidgetGroup(pParent), m_pWidget(NULL), m_pLayout(NULL) { }
+
+void WSingletonContainer::setup(QDomNode node, const SkinContext& context) {
+    setContentsMargins(0, 0, 0, 0);
+    m_pLayout = new QVBoxLayout();
+    m_pLayout->setSpacing(0);
+    m_pLayout->setContentsMargins(0, 0, 0, 0);
+    setLayout(m_pLayout);
+
+    QDomElement containerNode = node.toElement();
+    if (!context.hasNode(node, "ObjectName")) {
+        SKIN_WARNING(node, context)
+                << "Need objectName attribute for Singleton tag";
+        return;
+    }
+    QString objectName = context.selectString(node, "ObjectName");
+    m_pWidget = context.getSingletonWidget(objectName);
+    if (m_pWidget == NULL) {
+        SKIN_WARNING(node, context)
+                << "Asked for an unknown singleton widget:" << objectName;
+    }
+}
+
+void WSingletonContainer::showEvent(QShowEvent* event) {
+    Q_UNUSED(event);
+    if (m_pWidget) {
+        QWidget* parent = m_pWidget->parentWidget();
```

Comment:
Some comments would be helpfull here.
Does it steel the widget from Its original parent ... ?